src/main/java/com/vmware/vipclient/i18n/messages/service/ProductService.java

```diff
@@ -44,23 +51,39 @@ public JSONArray getSupportedLocalesFromRemoteVIP() {
 
     public List<Map> getAllComponentTranslation() {
         List<Map> list = new ArrayList<Map>();
-        Object[] locales = {};
-        Object[] components = {};
-        if (VIPCfg.getInstance().getMessageOrigin() == DataSourceEnum.VIP) {
-            locales = this.getSupportedLocalesFromRemoteVIP().toArray();
-            components = this.getComponentsFromRemoteVIP()
-                    .toArray();
-        }
-        for (Object locale : locales) {
-            for (Object component : components) {
-                dto.setComponent(((String) component).trim());
-                dto.setLocale(LocaleUtility.fmtToMappedLocale((String) locale).toString().trim());
-                Map<String, String> retMap = new ComponentService(dto).getMessages().getCachedData();
-                if (retMap != null) {
-                    list.add(retMap);
+        Map<String, String> locales = this.getLanguages(VIPCfg.getInstance().getMsgOriginsQueue().iterator());
```

Comment:
Don't need to pass msgSourceQueueIter because the invoked methods can get it from config.

Reply:
The iterator needs to be a parameter so that it can be used in the recursive method call (which I forgot to include -- I have just added it now)